crates/evm/evm/src/executors/corpus.rs

```diff
@@ -638,6 +687,303 @@ impl CorpusManager {
             function.abi_encode_input(&prev_inputs).map_err(|e| eyre!(e.to_string()))?.into();
         Ok(())
     }
+
+    // Sync Methods
+
+    /// Exports the new corpus entries to the master workers (id = 0) sync dir.
+    fn export(&self) -> eyre::Result<()> {
+        // Early return if no new entries or corpus dir not configured
+        if self.new_entry_indices.is_empty() || self.worker_dir.is_none() {
+            return Ok(());
+        }
+
+        let worker_dir = self.worker_dir.as_ref().unwrap();
+
+        // Master doesn't export (it only receives from others)
+        if self.id == 0 {
+            return Ok(());
+        }
+
+        let Some(master_sync_dir) = self
+            .config
+            .corpus_dir
+            .as_ref()
+            .map(|dir| dir.join(format!("{WORKER}0")).join(SYNC_DIR))
+        else {
+            return Ok(());
+        };
+
+        let mut exported = 0;
+        let corpus_dir = worker_dir.join(CORPUS_DIR);
+
+        for &index in &self.new_entry_indices {
+            if let Some(entry) = self.in_memory_corpus.get(index) {
+                let ext = self
+                    .config
+                    .corpus_gzip
+                    .then_some(format!("{JSON_EXTENSION}.gz"))
+                    .unwrap_or(JSON_EXTENSION.to_string());
+                let file_name = format!("{}-{}{ext}", entry.uuid, entry.timestamp);
+                let file_path = corpus_dir.join(&file_name);
+                let sync_path = master_sync_dir.join(&file_name);
+
+                let Ok(_) = foundry_common::fs::copy(file_path, sync_path) else {
```

Comment:
I wonder if would could symlink and only copy if it is going to be deleted or moved later.